website/docs/Use-Cases/Auto-Generation.md

```diff
@@ -62,6 +62,60 @@ Notes:
 - When `human_input_mode` is set to `"ALWAYS"`, the user proxy agent solicits human input every time a message is received; and the conversation stops when the human input is "exit", or when the received message is a termination message and no human input is provided.
 - When `human_input_mode` is set to `"TERMINATE"`, the user proxy agent solicits human input only when a termination message is received or the number of auto reply reaches `max_consecutive_auto_reply`.
 
+We also support the newly added [function calling capability of OpenAI's Chat Completions API](https://openai.com/blog/function-calling-and-other-api-updates?ref=upstract.com). One can pass in list of callable functions or class methods to `UserProxyAgent`, which corresponding to the description of functions passed to OpenAI's API.
+
+Example usage of the agents to solve a task with function calling feature:
+```python
+from flaml.autogen.agent import AssistantAgent, UserProxyAgent
+
+# put the descriptions of functions in config to be passed to OpenAI's API
+oai_config = {
+    "model": "gpt-4-0613",
+    "functions": [
+        {
+            "name": "execute_code",
+            "description": "Receive a list of python code or shell script and return the execution result.",
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "code_type": {
+                        "type": "string",
+                        "description": "Code type, 'python' or 'sh'.",
+                    },
+                    "code": {
+                        "type": "string",
+                        "description": "Valid Python code to execute.",
+                    }
+                },
+                "required": ["code_type", "code"],
+            },
+        }
+    ],
+    "function_call": "auto",
+}
+
+# create an AssistantAgent instance named "assistant"
+chatbot = AssistantAgent("assistant", config_list=config_list, **oai_config)
+
+# define your own function. Here we use a pre-defined '_execute_code' function from a UserProxyAgent instance
+def execute_code_function(code_type, code):
+    execute_code_function = UserProxyAgent(name="execute_code", work_dir="coding", use_docker=False)._execute_code
+    return execute_code_function([(code_type, code)])
+
+# create a UserProxyAgent instance named "user", the execute_code_function is passed
+user = UserProxyAgent(
+    "user",
+    human_input_mode="NEVER",
+    function_map={"execute_code": execute_code_function},
+)
+
+# start the conversation
+chatbot.receive(
+    "Draw a rocket and save to a file named 'rocket.svg'",
+    user,
+)
+```
```

Comment:
With the newly added content, I suggest adding subsections for "humna_input_mode" and "function_call" for better readability.